x-pack/platform/plugins/shared/lens/public/datasources/form_based/to_esql.test.ts

```diff
@@ -327,4 +327,48 @@ describe('to_esql', () => {
       `FROM myIndexPattern | WHERE order_date >= ?_tstart AND order_date <= ?_tend | STATS bucket_0_0 = COUNT(*) BY order_date = BUCKET(\`order_date\`, 30 minutes) | SORT order_date ASC`
     );
   });
+
+  it('should work with custom filters on the layer', () => {
+    const esql = getESQLForLayer(
+      [
+        [
+          '1',
+          {
+            operationType: 'date_histogram',
+            sourceField: 'order_date',
+            label: 'Date histogram',
+            dataType: 'date',
+            isBucketed: true,
+            interval: 'auto',
+          },
+        ],
+        [
+          '2',
+          {
+            operationType: 'count',
+            sourceField: 'records',
+            label: 'Count',
+            dataType: 'number',
+            isBucketed: false,
+            filter: {
+              query: 'geo.src:"US"',
+              language: 'kuery',
+            },
+          },
+        ],
+      ],
+      layer,
+      indexPattern,
+      uiSettings,
+      {
+        fromDate: '2021-01-01T00:00:00.000Z',
+        toDate: '2021-01-01T23:59:59.999Z',
+      },
+      new Date()
+    );
+
+    expect(esql?.esql).toEqual(
+      `FROM myIndexPattern | WHERE order_date >= ?_tstart AND order_date <= ?_tend | STATS bucket_0_0 = COUNT(*) WHERE KQL("geo.src:"US"") BY order_date = BUCKET(\`order_date\`, 30 minutes) | SORT order_date ASC`
```

Comment:
Why this doesn't reports the triple quotes for the `KQL` filter?
I was expecting this from the code

Reply:
fixed